### PR TITLE
os/drivers/wireless: Modify scanning result handler

### PR DESCRIPTION
**Pull request by SeonghoByeon**
- Remove unused scanning implementation in rtk_wifi_utils
- Modify scanning handler logic
- Fix memory leak after handling scanning data

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [8ca5532f66d075a38736e8672797181bda4b964e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8ca5532f66d075a38736e8672797181bda4b964e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cdc509ac406bc9c39126be5a7efd7a8e0b559029] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cdc509ac406bc9c39126be5a7efd7a8e0b559029] - Code Rule Check OK.

File: os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c [hunk 1]
```diff
@@ -772,91 +772,107 @@ static void print_scan_result(rtw_scan_result_t *record)
...
-extern int8_t wifi_scan_result_callback(wifi_utils_scan_list_s *scan_list);
+/* Scanning operation should be blocked mode */
+wifi_utils_scan_list_s *g_scan_list;
+int g_scan_num;
+static void _free_scanlist(void)
+{
+	while (g_scan_list) {
+		wifi_utils_scan_list_s *cur = g_scan_list;
+		g_scan_list = g_scan_list->next;
+		rtw_mfree(cur, sizeof(cur));
```
Comment (Taejun-Kwon) [inline os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c]: Null Check!

Comment (SeonghoByeon) [reply os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c]: Thanks for the comment. While loop itself already checks the nullity of g_scan_list. Therefore, (wifi_utils_scan_list_s *) cur becomes never NULL here.

Comment (Taejun-Kwon) [reply os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c]: Ok I missed it.

Comment (Taejun-Kwon) [reply os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c]: Ok I missed it.

File: os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c [hunk 2]
```diff
@@ -772,91 +772,107 @@ static void print_scan_result(rtw_scan_result_t *record)
...
-
-		if (i > 0) {
-			(_scan_list - 1)->next = _scan_list;
-			_scan_list->next = NULL;
+		if (g_scan_list == NULL) {
+			g_scan_list = scan_list;
+			g_scan_num = 1;
+		} else {
+			wifi_utils_scan_list_s *cur = g_scan_list;
+			int idx;
```
Comment (NilabhShant) [inline os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c]: declaring variable just before you need it is correct ?
OR, all variable should be declared as the function starts.

**Comment by seinfra** [comment]
Target : [ce06c03f86bb95c92db716ed1422bb0e83889a81] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ce06c03f86bb95c92db716ed1422bb0e83889a81] - Code Rule Check OK.

**Pull request merged**